src/nbl/CMakeLists.txt

```diff
@@ -406,31 +403,12 @@ macro(nbl_target_include_parallel_hashmap _trgt)
 endmacro()
 
 macro(nbl_target_link_openssl _trgt)
-	if(WIN32)
+	if(NOT (ANDROID AND "${CMAKE_HOST_SYSTEM_NAME}" STREQUAL "Windows")) # OpenSSL can't be built for Android platform using windows OS host
 		target_link_libraries(${_trgt} INTERFACE
-			"$<TARGET_PROPERTY:openssl_build,BINARY_DIR>/build/lib/libeay32.lib"
-			"$<TARGET_PROPERTY:openssl_build,BINARY_DIR>/build/lib/ssleay32.lib"
+			"$<TARGET_PROPERTY:OpenSSL,INTERFACE_LINK_LIBRARIES>"
 		)
-	elseif(ANDROID)
-	
-		if("${CMAKE_HOST_SYSTEM_NAME}" STREQUAL "Linux")
-			target_link_libraries(${_trgt} INTERFACE
-				"$<TARGET_PROPERTY:openssl_build,BINARY_DIR>/build/lib/libcrypto.a"
-				"$<TARGET_PROPERTY:openssl_build,BINARY_DIR>/build/lib/libssl.a"
-			)
-		#elseif("${CMAKE_HOST_SYSTEM_NAME}" STREQUAL "Windows")
-		# TODO - unify it
-		# 	     when openssl works with clang as windows host	
-		endif()
-	else()
-		target_link_libraries(${_trgt} INTERFACE
-				"$<TARGET_PROPERTY:openssl_build,BINARY_DIR>/build/lib/libcrypto.a"
-				"$<TARGET_PROPERTY:openssl_build,BINARY_DIR>/build/lib/libssl.a"
-			)
-	endif()
 
-	if(NOT (ANDROID AND "${CMAKE_HOST_SYSTEM_NAME}" STREQUAL "Windows"))
-		target_include_directories(${_trgt} PUBLIC "$<TARGET_PROPERTY:openssl_build,BINARY_DIR>/build/include")
+		target_include_directories(${_trgt} PUBLIC "$<TARGET_PROPERTY:OpenSSL,BINARY_DIR>/build/include")
```

Comment:
what does this path evaluate to?

Reply:
it is the same as it was, but OpenSSL is a IMPORTED library target - not custom target. The thing is that custom targets aren't allowed to be exported and because we have to reference it's properties by generator expression there would always be an error while exporting targets, hence the change that is a must.

Reply:
uh small mistake - INTERFACE, not IMPORTED